frontend/src/components/AppMap.vue

```diff
@@ -1,42 +1,109 @@
+      <div
+        ref="element"
+        class="map"
+        :style="{
+          width: width ? width + 'px' : '100%',
+          height: height ? height + 'px' : '100%',
+        }"
+      />
+    </div>
+
+    <!-- control set -->
+    <div class="controls">
+      <AppButton
+        v-tooltip="'Download current map view as PNG'"
+        :icon="faDownload"
+        :accent="true"
+        @click="download"
+        >Download Map</AppButton
+      >
+      <AppButton
+        v-tooltip="'Zoom out'"
+        :icon="faMinus"
+        @click="map?.zoomOut()"
+      />
+      <AppButton v-tooltip="'Zoom in'" :icon="faPlus" @click="map?.zoomIn()" />
+      <AppButton
+        v-tooltip="'Fit view to data'"
+        :icon="faCropSimple"
+        @click="fit"
+        >Fit</AppButton
+      >
+      <AppButton
+        v-tooltip="'View map in full screen'"
+        :icon="faExpand"
+        @click="fullscreen"
+        >Fullscreen</AppButton
+      >
+    </div>
+
+    <!-- top left legend -->
+    <Teleport v-if="showLegends && topLeftLegend" :to="topLeftLegend">
+      <div v-stop class="legend">
+        <slot name="top-left" />
+
+        <!-- scale key -->
+        <div class="steps">
+          <template
+            v-for="(step, index) of [...scale.steps].reverse()"
+            :key="index"
+          >
+            <svg viewBox="0 0 1 1">
+              <rect x="0" y="0" width="1" height="1" :fill="step.color" />
+            </svg>
+            <span>{{ formatValue(step.lower, min, max) }}</span>
+            <span>&ndash;</span>
+            <span>{{ formatValue(step.upper, min, max) }}</span>
+          </template>
+        </div>
       </div>
+    </Teleport>
 
-      <div class="steps">
-        <template v-for="(step, index) of legendSteps" :key="index">
-          <svg viewBox="0 0 1 1">
-            <rect x="0" y="0" width="1" height="1" :fill="step.color" />
-          </svg>
-          <span>{{ step.start }}</span>
-          <span>&ndash;</span>
-          <span>{{ step.end }}</span>
-        </template>
+    <!-- top right legend -->
+    <Teleport v-if="showLegends && topRightLegend" :to="topRightLegend">
+      <div v-stop class="legend">
+        <slot name="top-right" />
       </div>
-    </div>
-  </Teleport>
-
-  <Teleport v-if="popupTeleport && popupFeature" :to="popupTeleport">
-    <div class="popup">
-      <span>Name</span>
-      <span>{{ popupFeature.name }}</span>
-      <span>FIPS</span>
-      <span>{{ popupFeature.id }}</span>
-      <span>Value</span>
-      <span>{{ formatValue(values[popupFeature.id] || 0) }}</span>
-    </div>
-  </Teleport>
+    </Teleport>
+
+    <!-- bottom right legend -->
+    <Teleport
+      v-if="showLegends && bottomRightLegend"
+      :to="bottomRightLegend"
+      class="test"
+    >
+      <div v-stop class="legend">
+        <slot name="bottom-right" />
+
+        <!-- symbol key -->
+        <div v-if="Object.keys(symbols).length" class="symbols">
```

Comment:
Add legend slots for each corner (except one with attribution). Here is a new one showing the marker symbols.